poncho/src/poncho/package_serverize.py

```diff
@@ -14,226 +14,221 @@
 import ast
 import tarfile
 import hashlib
+import inspect
 
 shebang = "#! /usr/bin/env python3\n\n"
 
-vine_network_code = \
-'''
-import socket
-import json
-import os
-import sys
-import threading
-import queue
+def library_network_code():
+    import json
+    import os
+    import sys
+    import threading
+    import queue
 
-def remote_execute(func):
-    def remote_wrapper(event, q=None):
-        if q:
-            event = json.loads(event)
-        kwargs = event["fn_kwargs"]
-        args = event["fn_args"]
-        try:
-            response = {
-                "Result": func(*args, **kwargs),
-                "StatusCode": 200
-            }
-        except Exception as e:
-            response = { 
-                "Result": str(e),
-                "StatusCode": 500 
-            }
-        if not q:
-            return response
-        q.put(response)
-    return remote_wrapper
-    
-read, write = os.pipe() 
+    def remote_execute(func):
+        def remote_wrapper(event, q=None):
+            if q:
+                event = json.loads(event)
+            kwargs = event["fn_kwargs"]
+            args = event["fn_args"]
+            try:
+                response = {
+                    "Result": func(*args, **kwargs),
+                    "StatusCode": 200
+                }
+            except Exception as e:
+                response = { 
+                    "Result": str(e),
+                    "StatusCode": 500 
+                }
+            if not q:
```

Comment:
More natural:
```python
if q:
   q.put(response)

return response
```
In that way the function always returns the same and a potential caller does not have to think about that.

Reply:
The q is only used in threading execution modes, and Matt's upcoming PR is getting rid of the thread execution method (including this q variable). So his PR will be fixing this.